CreateTOC.ps1

```diff
@@ -0,0 +1,50 @@
+$scriptPath = Split-Path -Parent $MyInvocation.MyCommand.Definition
+$rootFolder = $scriptPath
+$outputFile = Join-Path $scriptPath "TOC.md"
+
+if (Test-Path $outputFile) {
+    Clear-Content $outputFile
+}
+
+function Format-FileName {
+    param (
+        [string]$fileName
+    )
+    $fileNameWithoutExtension = [System.IO.Path]::GetFileNameWithoutExtension($fileName)
+    $formattedName = $fileNameWithoutExtension -replace '-', ' ' 
+    return (Get-Culture).TextInfo.ToTitleCase($formattedName)
+}
+
+function Create-MarkdownLink {
+    param (
+        [string]$path,
+        [string]$relativePath
+    )
+    $fileName = Split-Path $path -Leaf
+    $formattedFileName = Format-FileName $fileName
+    $markdownLink = "- [$formattedFileName]($relativePath)"
+    Add-Content -Path $outputFile -Value $markdownLink
+}
+
+function Create-FolderHeader {
+    param (
+        [string]$folderPath
+    )
+    $relativeFolderPath = $folderPath.Substring($rootFolder.Length).Replace('\', '/')
+    $folderName = Split-Path $relativeFolderPath -Leaf
+    $header = "## $folderName"
+    Add-Content -Path $outputFile -Value ""
+    Add-Content -Path $outputFile -Value $header
+}
+
+Get-ChildItem -Path $rootFolder -Recurse -Directory | ForEach-Object {
+    $folderPath = $_.FullName
```

Comment:
very cool stuff, just one question: what is `$_`? Is it like a built-in iteration variable?

Reply:
I think so but I'm not a powershell expert

https://stackoverflow.com/questions/3494115/what-does-mean-in-powershell